tests/platform_tests/test_intf_fec.py

```diff
@@ -179,3 +140,57 @@ def skip_ber_counters_test(intf_status: dict) -> bool:
             pytest.fail("Pre-FEC and Post-FEC BER are not valid floats for interface {}, \
                     fec_pre_ber: {} fec_post_ber: {}"
                         .format(intf_name, fec_pre_ber, fec_post_ber))
+
+
+def get_fec_histogram(duthost, intf_name):
+    """
+    @Summary: Fetch FEC histogram for a given interface.
+    """
+    try:
+        logging.info("Get output of 'show interfaces counters fec-histogram {}'".format(intf_name))
+        fec_hist = duthost.show_and_parse("show interfaces counters fec-histogram {}".format(intf_name))
+    except Exception as e:
+        logging.error("Failed to execute 'show interfaces counters fec-histogram {}': {}".format(intf_name, e))
+        pytest.fail("Command 'show interfaces counters fec-histogram {}' not found \
+                or failed: {}".format(intf_name, str(e)))
+        return None
+
+    if not fec_hist:
+        pytest.fail("No FEC histogram data found for interface {}".format(intf_name))
+
+    logging.info("FEC histogram for interface {}: {}".format(intf_name, fec_hist))
+    return fec_hist
+
+
+def validate_fec_histogram(duthost, intf_name):
+    """
+    @Summary: Validate FEC histogram critical bins for any errors. Fail the test if bin value > 0
+    """
+
+    fec_hist = get_fec_histogram(duthost, intf_name)
+    if not fec_hist:
+        pytest.fail("FEC histogram data not found for interface {}".format(intf_name))
+
+    critical_bins = range(7, 16)
+    for bin_index in critical_bins:
+        bin_value = int(fec_hist[bin_index].get('codewords', 0))
+        if bin_value > 0:
+            pytest.fail("FEC histogram bin {} has errors for interface {}: {}"
+                        .format(bin_index, intf_name, bin_value))
```

Comment:
wouldn't it be better to accumulate all the failures and then print it to the user?